packages/nextjs/src/config/withSentryConfig.ts

```diff
@@ -5,26 +5,32 @@ import type {
   NextConfigObject,
   NextConfigObjectWithSentry,
   SentryWebpackPluginOptions,
+  UserSentryOptions,
 } from './types';
 
 /**
  * Add Sentry options to the config to be exported from the user's `next.config.js` file.
  *
  * @param exportedUserNextConfig The existing config to be exported prior to adding Sentry
  * @param userSentryWebpackPluginOptions Configuration for SentryWebpackPlugin
+ * @param sentryWizardAddon Addons from automatic config.next.js merging from Sentry Wizard
  * @returns The modified config to be exported
  */
 export function withSentryConfig(
   exportedUserNextConfig: ExportedNextConfig = {},
   userSentryWebpackPluginOptions: Partial<SentryWebpackPluginOptions> = {},
+  sentryWizardAddon?: UserSentryOptions,
 ): NextConfigFunction | NextConfigObject {
   return function (phase: string, defaults: { defaultConfig: NextConfigObject }): NextConfigObject {
+    let userNextConfigObject;
     if (typeof exportedUserNextConfig === 'function') {
-      const userNextConfigObject = exportedUserNextConfig(phase, defaults);
-      return getFinalConfigObject(phase, userNextConfigObject, userSentryWebpackPluginOptions);
+      userNextConfigObject = exportedUserNextConfig(phase, defaults);
```

Comment:
@roggenkemper Nit: you could make this more concise:
`const userNextConfigObject typeof exportedUserNextConfig === 'function' ? exportedUserNextConfig(phase, defaults) : exportedUserNextConfig`;